src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -274,6 +274,9 @@ internal static bool TryParse(ReadOnlySpan<char> s, out char result)
         /// This determines whether the character is in the range 'A' through 'Z', inclusive,
         /// 'a' through 'z', inclusive, or '0' through '9', inclusive.
         /// </remarks>
+        // Use of the non-short-circuiting logical OR operator (|) here is an intentional optimization.
+        // It avoids a low-level branch instruction.  (In other words, the overhead of branching
+        // is greater than the cost of simply executing IsBetween.)
```

Comment:
The comment looks too verbose for users who can understand it. The position is also questionable with xml doc.

I'd suggest a trailing comment on the same line, just saying "use arithmetic or to avoid branching" or similar.